### PR TITLE
 Mark DTLS records as read when we have finished with them

### PR DESCRIPTION
**Pull request by mattcaswell**
The TLS code marks records as read when its finished using a record. The DTLS code did
not do that. However SSL_has_pending() relies on it. So we should make DTLS consistent.

Also we shouldn't set TCP_NODELAY on a UDP socket

### CODE REVIEW

**Comment by mattcaswell** [comment]
Found these issues while trying to fix #4929. Now I can get back to that one!

**Comment by mattcaswell** [comment]
Ping?

File: ssl/record/rec_layer_d1.c [hunk 1]
```diff
@@ -495,21 +496,33 @@ int dtls1_read_bytes(SSL *s, int type, int *recvd_type, unsigned char *buf,
...
+        }
 
         if (len > SSL3_RECORD_get_length(rr))
             n = SSL3_RECORD_get_length(rr);
         else
             n = len;
 
         memcpy(buf, &(SSL3_RECORD_get_data(rr)[SSL3_RECORD_get_off(rr)]), n);
-        if (!peek) {
+        if (peek) {
```
Comment (richsalz) [inline ssl/record/rec_layer_d1.c]: ah, inverting the test was a good catch.

**Comment by mattcaswell** [comment]
Pushed - thanks. Could you also review the 1.1.0 version? #6160.

**Pull request closed without merging**